### PR TITLE
CHAT-705 Number of search results in user search not limited

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

File: application/src/main/webapp/js/chat.js [hunk 1]
```diff
@@ -1114,6 +1114,7 @@ var chatApplication = new ChatApplication();
               data: {
                 "filter": query,
                 "user": chatApplication.username,
+                "limit": 10,
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: It means we will never display more than 10 users in the dropdown list, is it what we want (just asking to be sure) ?

File: server/src/main/java/org/exoplatform/chat/server/ChatServer.java [hunk 2]
```diff
@@ -891,19 +891,19 @@ public ChatServer()
...
-    List<UserBean> users;
-    if (room != null && !"".equals(room)) {
-      users = userService.getUsers(room, dbName);
-    } else {
-      users = userService.getUsers(filter, true, dbName);
-    }
+    int limit_ = 0;
+    try {
+      limit_ = Integer.valueOf(limit);
+    } catch (NumberFormatException e) {}
```
Comment (thomasdelhomenie) [inline server/src/main/java/org/exoplatform/chat/server/ChatServer.java]: Avoid empty catch. In this case, the input parameters are not valid, so it should return an HTTP error 400 IMO.

File: application/src/main/webapp/js/chat.js (hunk 1 above)
Comment (trongtt) [inline application/src/main/webapp/js/chat.js]: No, this number is only for returned items from server each time we make a search request. In client side, it maintains its own list of all items returned so far and display items basing on its criteria search.

File: server/src/main/java/org/exoplatform/chat/server/ChatServer.java (hunk 2 above)
Comment (trongtt) [inline server/src/main/java/org/exoplatform/chat/server/ChatServer.java]: Done

File: server/src/main/java/org/exoplatform/chat/server/ChatServer.java [hunk 3]
```diff
@@ -891,20 +891,22 @@ public ChatServer()
...
-    List<UserBean> users;
-    if (room != null && !"".equals(room)) {
-      users = userService.getUsers(room, dbName);
-    } else {
-      users = userService.getUsers(filter, true, dbName);
+    int limit_ = 0;
+    try {
+      limit_ = Integer.valueOf(limit);
+    } catch (NumberFormatException e) {
+      return Response.status(400).content("Bad Request !");
```
Comment (thomasdelhomenie) [inline server/src/main/java/org/exoplatform/chat/server/ChatServer.java]: We should explain why it is a bad request in the message.

Comment (trongtt) [inline server/src/main/java/org/exoplatform/chat/server/ChatServer.java]: Done.

**Pull request merged**